app/controllers/sys_controller.rb

```diff
@@ -33,45 +33,13 @@ class SysController < ActionController::Base
 
   before_action :check_enabled
   before_action :require_basic_auth, only: [:repo_auth]
-  before_action :find_project, only: [:update_required_storage]
-  before_action :find_repository_with_storage, only: [:update_required_storage]
 
   def projects
-    p = Project.active.has_module(:repository)
-        .includes(:repository)
-        .references(:repositories)
-        .order(Arel.sql("identifier"))
-    respond_to do |format|
-      format.json do
-        render json: p.to_json(include: :repository)
-      end
-      format.any(:html, :xml) do
-        render xml: p.to_xml(include: :repository), content_type: Mime[:xml]
-      end
-    end
+    render text: "", status: :gone
```

Comment:
With the routes dispatcher already returning a 410 I think those stub methods in here can be completely removed.

Reply:
Good catch, I first added those before adding the routes. Will take care of it